### PR TITLE
Feature/e2e tests

### PR DESCRIPTION
**Pull request by mikemoody-amido**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -186,7 +186,7 @@ workflows:
             branches:
               only: release
       - assume-role-staging:
-          context: api-assume-role-staging-context
+          context: api-assume-role-housing-staging-context
```
Comment (evak6979) [inline .circleci/config.yml:189]: I really wonder why this had chosen the wrong role to begin with. Did this come from an outdated baseapi?

File: HousingSearchApi.Tests/V1/Helper/ElasticSearchFixture.cs [hunk 2]
```diff
@@ -1,37 +1,96 @@
...
                 if (null != _factory)
                     _factory.Dispose();
                 _disposed = true;
             }
         }
+
+        private void WaitForESInstance(IElasticClient elasticSearchClient)
+        {
+            Exception ex = null;
+            var timeout = DateTime.UtcNow.AddSeconds(10); // 5 second timeout (make configurable?)
```
Comment (evak6979) [inline HousingSearchApi.Tests/V1/Helper/ElasticSearchFixture.cs:61]: I don't think it's worth making configurable. But perhaps adding a comment to explain why we have this delay to begin with.

File: HousingSearchApi.Tests/V1/Helper/TestDataHelper.cs [hunk 3]
```diff
@@ -1,38 +1,64 @@
...
+            return persons;
         }
 
         private static List<QueryablePerson> CreateQueryablePerson()
         {
             var listOfPersons = new List<QueryablePerson>();
             var random = new Random();
+            var fixture = new Fixture();
+
+            // Make sure there are 10 of each surname first
```
Comment (evak6979) [inline HousingSearchApi.Tests/V1/Helper/TestDataHelper.cs:48]: Why ten though?

**Review by evak6979 — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (mikemoody-amido) [reply .circleci/config.yml:189]: Actually as the search was one of the first ones we built, I think that the role name changed after we first set up the project.

File: HousingSearchApi.Tests/V1/Helper/TestDataHelper.cs (hunk 3 above)
Comment (mikemoody-amido) [reply HousingSearchApi.Tests/V1/Helper/TestDataHelper.cs:48]: Well the page size test uses 5 so it was just making sure that there would always be more than that.

File: HousingSearchApi.Tests/V1/Helper/ElasticSearchFixture.cs (hunk 2 above)
Comment (mikemoody-amido) [reply HousingSearchApi.Tests/V1/Helper/ElasticSearchFixture.cs]: Updated comment

**Pull request merged**